boto3/session.py

```diff
@@ -280,6 +280,10 @@ def _register_default_handlers(self):
             boto3.utils.lazy_call(
                 'boto3.s3.inject.inject_s3_transfer_methods'))
```

Comment:
On a complete side note, we should look into adding ``upload_file`` and ``download_file`` to the Bucket resource.